packages/custom_lint_core/lib/src/type_checker.dart

```diff
@@ -221,7 +221,10 @@ abstract class TypeChecker {
   ///
   /// This only takes into account the *extends* hierarchy. If you wish
   /// to check mixins and interfaces, use [isAssignableFromType].
-  bool isSuperTypeOf(DartType staticType) => isSuperOf(staticType.element!);
+  bool isSuperTypeOf(DartType staticType) {
+    final element = staticType.element;
```

Comment:
Could you add a check here? https://github.com/invertase/dart_custom_lint/blob/main/packages/custom_lint_core/test/type_checker_test.dart

Reply:
Sure!